pom.xml

```diff
@@ -159,6 +159,12 @@
             <version>2.2</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>matrix-project</artifactId>
```

Comment:
Why?

Reply:
because of:
`Caused by: java.io.IOException: Jenkins Git plugin version 4.2.2 failed to load.
 - Matrix Project Plugin version 1.4.1 is older than required. To fix, install version 1.14 or later.`
😞

Reply:
Well the BOM defines a version for `matrix-project` so perhaps you just need an explicit dependency, but without `version`?

Reply:
I would also add a comment here explaining why the dependency exists, something like this:

```
<!-- Required because git plugin optionally depends on a recent version of matrix-project, and if we do not depend on matrix-project here, an old version of matrix-project is installed automatically as a detached plugin, causing git plugin to fail to load. -->
```